src/main/java/Duke.java

```diff
@@ -1,10 +1,115 @@
+import java.util.Scanner;
+
 public class Duke {
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Scanner in = new Scanner(System.in);
+        Task[] tasks = new Task[100];
+        int currentIndex = 0;
+
+        printHelloStatement();
+        while (true) {
+            String input;
+            input = in.nextLine();
+            if (input.equals("bye")) {
+                printByeStatement();
+                break;
+            } else if (input.equals("list")) {
+                printAllTasks(tasks);
+            } else if (input.startsWith("mark")) {
+                String[] temp = input.split(" ", 2);
+                int taskIndex = Integer.parseInt(temp[1]);
+                Task curTask = tasks[taskIndex - 1];
+                curTask.markAsDone();
+                printTaskStatusStatement(curTask,"mark");
+            } else if (input.startsWith("unmark")) {
+                String[] temp = input.split(" ", 2);
+                int taskIndex = Integer.parseInt(temp[1]);
+                Task curTask = tasks[taskIndex - 1];
+                curTask.unmarkAsDone();
+                printTaskStatusStatement(curTask,"unmark");
+            } else if (input.startsWith("todo")){
+                String[] temp = input.split(" ", 2);
+                String description = temp[1];
+                ToDo todo = new ToDo(currentIndex + 1, description);
+                tasks[currentIndex] = todo;
+                currentIndex++;
+                printTaskAddedStatement(currentIndex, todo);
+            }
+            else if (input.startsWith("deadline")){
+                String[] temp1 = input.split(" /by ", 2);
+                String by = temp1[1];
+                String[] temp2 = temp1[0].split(" ", 2);
+                String description = temp2[1];
+                Deadline deadline = new Deadline(currentIndex + 1, description, by);
+                tasks[currentIndex] = deadline;
+                currentIndex++;
+                printTaskAddedStatement(currentIndex, deadline);
+            }
```

Comment:
Perhaps it might be useful to indicate the intended format input via some comments?
So that we can understand why you are splitting through `"/by"` and what the outcome is supposed to be.